### PR TITLE
Fix idententity packet in AC

### PR DESCRIPTION
**Pull request by howard-stearns**
Fixes crash in AC caused by https://github.com/highfidelity/hifi/pull/9227

### CODE REVIEW

File: libraries/avatars/src/AvatarHashMap.cpp [hunk 1]
```diff
@@ -134,11 +134,16 @@ void AvatarHashMap::processAvatarIdentityPacket(QSharedPointer<ReceivedMessage>
...
     static auto EMPTY = QUuid();
-    if (identity.uuid == _avatarHash.value(EMPTY)->getSessionUUID()) {
-        // We add MyAvatar to _avatarHash with an empty UUID. Code relies on this. In order to correctly handle an
-        // identity packet for ourself (such as when we are assigned a sessionDisplayName by the mixer upon joining),
-        // we make things match here.
-        identity.uuid = EMPTY;
+
+    {
+        QWriteLocker locker(&_hashLock);
+        auto me = _avatarHash.find(EMPTY);
```
Comment (birarda) [inline libraries/avatars/src/AvatarHashMap.cpp]: if you're only doing a find you probably just need a QReadLocker

**Comment by howard-stearns** [comment]
smoke test plan:
- Run default home content and make sure you can hear background sounds.
- In console, while connected to the above local domain, make sure MyAvatar.sessionDisplayName is not the empty string.

File: libraries/avatars/src/AvatarHashMap.cpp (hunk 1 above)
Comment (howard-stearns) [inline libraries/avatars/src/AvatarHashMap.cpp]: arrgh...; Doh!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f4faec15ff7aef8e0971e12dc47ee3e158f52fd available at: https://gist.github.com/bbf864c43f6e5e54d9f5e8f3c1e24a44<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9248-3f4faec.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9248-3f4faec.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 10
QA Pass
Master (5826) MyAvatar.sessionDisplayName retruns anonymous_1 (or my display name if set)
This PR (9248) MyAvatar.sessionDisplayName returns same as above.
General smoke test passes

Master exhibits constant connect/disco cycle for scripted agent in sandbox default content (thus no sounds)
This PR exhibits normal working for scripted agent (thus sound works).

**Comment by hifi-gustavo** [comment]
Build console output for commit: a6e353d75f034005437297e3eb3aa9c21d49b743 available at: https://gist.github.com/bbf864c43f6e5e54d9f5e8f3c1e24a44<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9248-a6e353d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9248-a6e353d.dmg

**Pull request merged**